specification/communication/data-plane/Rooms/preview/2022-02-01-preview/communicationservicesrooms.json

```diff
@@ -0,0 +1,707 @@
+          },
+          {
+            "in": "body",
+            "name": "removeParticipantsRequest",
+            "description": "Participants in a room to be removed.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/RemoveParticipantsRequest"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The participants were successfully deleted. The action returns an updated `Room`.",
+            "schema": {
+              "$ref": "#/definitions/RoomModel"
+            }
+          },
+          "default": {
+            "description": "Error response",
+            "schema": {
+              "$ref": "#/definitions/CommunicationErrorResponse"
+            },
+            "headers": {
+              "x-ms-error-code": {
+                "x-ms-client-name": "ErrorCode",
+                "type": "string"
+              }
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Remove participant": {
+            "$ref": "./examples/Rooms_RemoveParticipants.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "CommunicationErrorResponse": {
+      "$ref": "../../../Common/stable/2021-03-07/common.json#/definitions/CommunicationErrorResponse"
+    },
+    "CreateRoomRequest": {
+      "description": "Request payload for creating new room.",
+      "type": "object",
+      "properties": {
+        "validFrom": {
+          "format": "date-time",
+          "description": "The timestamp from when the room is open for joining. The timestamp is in RFC3339 format: `yyyy-MM-ddTHH:mm:ssZ`.",
+          "type": "string",
+          "example": "2021-09-07T07:55:41Z"
+        },
+        "validUntil": {
+          "format": "date-time",
+          "description": "The timestamp from when the room can no longer be joined. The timestamp is in RFC3339 format: `yyyy-MM-ddTHH:mm:ssZ`.",
+          "type": "string",
+          "example": "2020-09-08T15:55:41Z"
+        },
+        "roomOpen": {
+          "description": "Flag to specify if the room is to be an open room or closed room.",
+          "type": "boolean",
+          "example": false
+        },
+        "participants": {
+          "description": "(Optional) Collection of identities invited to the room.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RoomParticipant"
+          },
+          "x-ms-identifiers": []
+        }
+      }
+    },
+    "RoomParticipant": {
+      "description": "A participant of the room.",
+      "required": [
+        "communicationIdentifier"
+      ],
+      "type": "object",
+      "properties": {
+        "communicationIdentifier": {
+          "$ref": "#/definitions/CommunicationIdentifierModel"
+        },
+        "role": {
+          "description": "Role Name.",
+          "type": "string"
+        }
+      }
+    },
+    "CommunicationIdentifierModel": {
```

Comment:
I'm a bit surprised by this definition.
If I understand correctly, this is trying to keep the data shape of `CommunicaitonUserIdentifierModel` from the communication/common package, but restricting the enum to just the ACS user.

Is there precedent for this? If we need this enum-subset type here (and presumably elsewhere), I think we should still define it in the communication/common package.

Reply:
Yeah, I was also surprised by this. But apparently, there is a precedent - see e.g. chat:
https://docs.microsoft.com/en-us/rest/api/communication/chat/chat/create-chat-thread

Reply:
Aha...the `rawId` is optional
> "Raw Id of the identifier. Optional in requests, required in responses."

https://github.com/Azure/azure-rest-api-specs/blob/03a43aa61a1e02d84c902b2c2ed1bd341833dcfa/specification/communication/data-plane/Common/stable/2021-03-07/common.json#L73-L94

Reply:
@prprabhu-ms 
> If I understand correctly, this is trying to keep the data shape of CommunicaitonUserIdentifierModel from the communication/common package, but restricting the enum to just the ACS user.

It's in the shape of CommunicationIdentifier, not CommunicationUserIdentifier...so it doesn't restrict it only to ACS users.

Reply:
My comment was about the *definition* of this model here.

Chat just references the common model: https://github.com/Azure/azure-rest-api-specs/blob/03a43aa61a1e02d84c902b2c2ed1bd341833dcfa/specification/communication/data-plane/Chat/stable/2021-09-07/communicationserviceschat.json#L1200

Reply:
They reference the common model on the line 522 and 527. They're just wrapping it in a parent type that also has a role attached to it.

Reply:
Great question. Let me explain why we are not referring to the common CommunicationIdentifierModel.
At this point in time Rooms do not support Microsoft Teams User Identifier and Phone number identifier. We do have this in pipeline. 
In order to refer only to the Communication User Identifier which Rooms support right now, we have defined CommunicationIdentifierModel at our spec level rather than using the common one which includes all 3 identifiers and referring to CommunicationUserIdentifierModel from the communication common. 
Hope that clarifies.

Reply:
I get your point @prprabhu-ms, you're right.
@elavarasidc Yes, all clear. Just pehaps one note - make sure that your specification of the `CommunicationIdentifierModel` never diverges from the common-level spec so that you'll be always able to make the transition back.

Reply:
Will they be able to make the transition though? That's my main worry.

`CommunicationIdentifierModel` is a polymorphic type. I'm not sure if adding more variants to a polymorphic type is considered a breaking change. That's a question for the API stewardship council.

Reply:
> I get your point @prprabhu-ms, you're right. @elavarasidc Yes, all clear. Just perhaps one note - make sure that your specification of the `CommunicationIdentifierModel` never diverges from the common-level spec so that you'll be always able to make the transition back.

@petrsvihlik yes, we the plan is to make sure we align with the common CommunicationIdentifierModel.